uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -250,35 +230,66 @@ public void handleFirstRun() {
     @Nullable
     public String getAccountUpn() {
         Logger.i(TAG, "Get Account Upn..");
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonGetWpjUpn";
+        final String textId = "UPN";
+        return basicButtonHandler(resourceButtonId,textId);
+    }
+
+    @Nullable
+    public String getDeviceState() {
+        Logger.i(TAG, "Get Device State ..");
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonDeviceState";
+        final String textId = "DeviceState";
+        return basicButtonHandler(resourceButtonId,textId);
+    }
+
+    @Nullable
+    public boolean isDeviceShared() {
+        Logger.i(TAG, "Check if device is shared..");
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonIsDeviceShared";
+        final String textId = "DeviceShared";
+        final String isDeviceSharedText= basicButtonHandler(resourceButtonId,textId);
+        return "Device is shared".equalsIgnoreCase(isDeviceSharedText);
+    }
+
+    @Nullable
+    public String wpjLeave() {
+        Logger.i(TAG, "Wpj Leave ..");
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonLeave";
+        final String textId = "wpjLeave";
+        return basicButtonHandler(resourceButtonId,textId);
+    }
+
+    @Nullable
+    private final String basicButtonHandler(@NonNull final String resourceButtonId, @NonNull final String textId){
         launch(); // launch Broker Host app
 
         if (shouldHandleFirstRun) {
             handleFirstRun(); // handle first run experience
         }
 
-        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/buttonGetWpjUpn");
+        UiAutomatorUtils.handleButtonClick(resourceButtonId);
 
-        // Look for the UPN dialog box
-        final UiObject showUpnDialog = UiAutomatorUtils.obtainUiObjectWithResourceId(
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
                 "android:id/message"
         );
-
-        Assert.assertTrue(showUpnDialog.exists());
+        dialogBox.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+        Assert.assertTrue(dialogBox.exists());
 
         try {
-            // Obtain the text on the UPN dialog box
-            final String[] upnDialogTextParts = showUpnDialog.getText().split(":");
-
-            // get the UPN if it is there, else return null (in case of error)
-            if ("UPN".equalsIgnoreCase(upnDialogTextParts[0])) {
-                return upnDialogTextParts[1];
+            // get the textId if it is there, else return null (in case of error)
+            final String[] dialogBoxText = dialogBox.getText().split(":");
+            // look for the textId if present
+            if (dialogBoxText[0].equalsIgnoreCase(textId)) {
```

Comment:
Since textId is NonNull, you may want to flip this:

`textId.equalsIgnoreCase(dialogBoxText[0])`

Reply:
948cdb496146cb7fe9ad30c038ac337726928722